staging/src/k8s.io/apiserver/pkg/util/flowcontrol/fairqueuing/queueset/fifo_list_test.go

```diff
@@ -148,6 +148,41 @@ func TestFIFOWithRemoveIsIdempotent(t *testing.T) {
 	verifyOrder(t, orderExpected, remainingRequests)
 }
 
+func TestFIFOWidth(t *testing.T) {
+	list := newRequestFIFO()
+
+	newRequest := func(width uint) *request {
+		return &request{width: width}
+	}
+	arrival := []*request{newRequest(1), newRequest(2), newRequest(3)}
+	removeFn := make([]removeFromFIFOFunc, 0)
+
+	width := 0
+	for i := range arrival {
+		removeFn = append(removeFn, list.Enqueue(arrival[i]))
+
+		width += i + 1
+		if list.Width() != width {
+			t.Errorf("Expected width: %d, but got: %d", width, list.Width())
+		}
+	}
+
```

Comment:
can we also add a test for another form of idempotency - if we call both `RemoveFn` and `Dequeue` on the same request object, we expect the width to be decreased once.

Reply:
done